msvc w/o

```
C:/PROGRA~2/MICROS~1/2019/COMMUN~1/VC/Tools/MSVC/1429~1.301/bin/Hostx64/x64/cl.exe  /nologo /TP -DNOMINMAX -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_FOREACH -DQT_POSITIONING_LIB -DQT_PRINTSUPPORT_LIB -DQT_SERIALPORT_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_USE_QSTRINGBUILDER -DQT_WEBKITWIDGETS_LIB -DQT_WEBKIT_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DSIP_VERSION=0x060101 -DWITH_QTWEBKIT -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS -D_HAS_AUTO_PTR_ETC=1 -D_USE_MATH_DEFINES -Dqgis_analysis_EXPORTS -ID:/src/osgeo4w/src/qgis-dev/osgeo4w/build/src/analysis/qgis_analysis_autogen/include -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/mesh -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/raster -ID:/src/osgeo4w/src/qgis-dev/osgeo4w/build -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/georeferencing -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/interpolation -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/network -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/processing -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/vector -ID:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/vector/geometry_checker -ID:/src/osgeo4w/src/qgis-dev/osgeo4w/build/src/analysis -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core -ID:/src/osgeo4w/src/qgis-dev/osgeo4w/build/src/core -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/3d -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/annotations -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/auth -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/browser -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/callouts -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/classification -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/diagram -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/dxf -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/editform -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/effects -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/elevation -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/expression -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/externalstorage -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/fieldformatter -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/geometry -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/geocms -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/geocms/geonode -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/geocoding -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/gps -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/labeling -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/layertree -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/layout -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/locator -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/maprenderer -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/mesh -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/metadata -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/network -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/numericformats -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/pal -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/plot -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/pointcloud -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/pointcloud/expression -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/processing -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/processing/models -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/proj -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/project -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/arcgis -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/memory -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/gdal -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/ogr -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/meshmemory -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/raster -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/renderer -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/scalebar -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/settings -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/symbology -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/textrenderer -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/validity -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/vector -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/vectortile -ID:/src/osgeo4w/src/qgis-dev/qgis/external -ID:/src/osgeo4w/src/qgis-dev/qgis/external/nlohmann -ID:/src/osgeo4w/src/qgis-dev/qgis/external/kdbush/include -ID:/src/osgeo4w/src/qgis-dev/qgis/external/nmea -ID:/src/osgeo4w/src/qgis-dev/qgis/external/rtree/include -ID:/src/osgeo4w/src/qgis-dev/qgis/external/meshOptimizer -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/ept -ID:/src/osgeo4w/src/qgis-dev/qgis/src/core/providers/copc -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/include -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtCore -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/./mkspecs/win32-msvc -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtGui -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtANGLE -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtXml -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtWidgets -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtSvg -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtPrintSupport -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtNetwork -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtSql -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtConcurrent -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/gdal-dev/include -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtCrypto -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/qt5keychain -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtSerialPort -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtWebKitWidgets -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtWebKit -external:ID:/src/osgeo4w/src/qgis-dev/osgeo4w/osgeo4w/apps/Qt5/include/QtPositioning -external:W0 /DWIN32 /D_WINDOWS /W3 /GR /EHsc  /wd4091  /wd4100  /wd4127  /wd4190  /wd4231  /wd4244  /wd4251  /wd4267  /wd4275  /wd4290  /wd4456  /wd4457  /wd4458  /wd4505  /wd4510  /wd4512  /wd4610  /wd4706  /wd4714  /wd4800  /wd4996  /utf-8 /std:c++17 /MD /Z7 /MP /Od /D NDEBUG -std:c++17 /showIncludes /Fosrc/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmpackage.cpp.obj /Fdsrc/analysis/CMakeFiles/qgis_analysis.dir/ /FS -c D:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/processing/qgsalgorithmpackage.cpp
D:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/processing/qgsalgorithmpackage.cpp(193): error C3493: 'maxRecursion' cannot be implicitly captured because no default capture mode has been specified
D:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/processing/qgsalgorithmpackage.cpp(271): error C3493: 'maxRecursion' cannot be implicitly captured because no default capture mode has been specified
D:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/processing/qgsalgorithmpackage.cpp(291): error C2064: term does not evaluate to a function taking 3 arguments
D:/src/osgeo4w/src/qgis-dev/qgis/src/analysis/processing/qgsalgorithmpackage.cpp(293): error C2064: term does not evaluate to a function taking 3 arguments
```

vs clang++ w/

```
2022-08-11T15:16:39.9167735Z /usr/lib64/ccache/clang++ -DQGIS_DISABLE_DEPRECATED -DQT_CONCURRENT_LIB -DQT_CORE5COMPAT_LIB -DQT_CORE_LIB -DQT_DEPRECATED_WARNINGS -DQT_DISABLE_DEPRECATED_BEFORE=0x050800 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_FOREACH -DQT_PRINTSUPPORT_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR -Dqgis_analysis_EXPORTS -I/root/QGIS/build/src/analysis/qgis_analysis_autogen/include -I/root/QGIS/src/analysis/mesh -I/root/QGIS/src/analysis/raster -I/root/QGIS/build -I/root/QGIS/src/analysis -I/root/QGIS/src/analysis/georeferencing -I/root/QGIS/src/analysis/interpolation -I/root/QGIS/src/analysis/network -I/root/QGIS/src/analysis/processing -I/root/QGIS/src/analysis/vector -I/root/QGIS/src/analysis/vector/geometry_checker -I/root/QGIS/build/src/analysis -I/root/QGIS/src/core -I/root/QGIS/build/src/core -I/root/QGIS/src/core/3d -I/root/QGIS/src/core/annotations -I/root/QGIS/src/core/auth -I/root/QGIS/src/core/browser -I/root/QGIS/src/core/callouts -I/root/QGIS/src/core/classification -I/root/QGIS/src/core/diagram -I/root/QGIS/src/core/dxf -I/root/QGIS/src/core/editform -I/root/QGIS/src/core/effects -I/root/QGIS/src/core/elevation -I/root/QGIS/src/core/expression -I/root/QGIS/src/core/externalstorage -I/root/QGIS/src/core/fieldformatter -I/root/QGIS/src/core/geometry -I/root/QGIS/src/core/geocms -I/root/QGIS/src/core/geocms/geonode -I/root/QGIS/src/core/geocoding -I/root/QGIS/src/core/gps -I/root/QGIS/src/core/labeling -I/root/QGIS/src/core/layertree -I/root/QGIS/src/core/layout -I/root/QGIS/src/core/locator -I/root/QGIS/src/core/maprenderer -I/root/QGIS/src/core/mesh -I/root/QGIS/src/core/metadata -I/root/QGIS/src/core/network -I/root/QGIS/src/core/numericformats -I/root/QGIS/src/core/pal -I/root/QGIS/src/core/plot -I/root/QGIS/src/core/pointcloud -I/root/QGIS/src/core/pointcloud/expression -I/root/QGIS/src/core/processing -I/root/QGIS/src/core/processing/models -I/root/QGIS/src/core/proj -I/root/QGIS/src/core/project -I/root/QGIS/src/core/providers -I/root/QGIS/src/core/providers/arcgis -I/root/QGIS/src/core/providers/memory -I/root/QGIS/src/core/providers/gdal -I/root/QGIS/src/core/providers/ogr -I/root/QGIS/src/core/providers/meshmemory -I/root/QGIS/src/core/raster -I/root/QGIS/src/core/renderer -I/root/QGIS/src/core/scalebar -I/root/QGIS/src/core/settings -I/root/QGIS/src/core/symbology -I/root/QGIS/src/core/textrenderer -I/root/QGIS/src/core/validity -I/root/QGIS/src/core/vector -I/root/QGIS/src/core/vectortile -I/root/QGIS/external -I/root/QGIS/external/nlohmann -I/root/QGIS/external/kdbush/include -I/root/QGIS/external/nmea -I/root/QGIS/external/rtree/include -I/root/QGIS/external/meshOptimizer -I/root/QGIS/src/core/providers/ept -I/root/QGIS/src/core/providers/copc -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtDBus -isystem /usr/include/qt6/QtXml -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtPrintSupport -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtSql -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/gdal -isystem /usr/local/include/Qca-qt6/QtCrypto -isystem /usr/local/include/qt6keychain -isystem /usr/lib64/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtCore5Compat -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Werror -Wnon-virtual-dtor -Wno-redundant-move -Wno-misleading-indentation -Wreturn-type-c-linkage -Woverloaded-virtual -Wimplicit-fallthrough -Wrange-loop-construct -Qunused-arguments -O3 -DNDEBUG -fPIC -fvisibility=hidden -std=gnu++17 -MD -MT src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmpackage.cpp.o -MF src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmpackage.cpp.o.d -o src/analysis/CMakeFiles/qgis_analysis.dir/processing/qgsalgorithmpackage.cpp.o -c /root/QGIS/src/analysis/processing/qgsalgorithmpackage.cpp
2022-08-11T15:16:39.9185419Z /root/QGIS/src/analysis/processing/qgsalgorithmpackage.cpp:138:84: error: lambda capture 'maxRecursion' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
2022-08-11T15:16:39.9188475Z       const auto findReferenced = [ &project, &feedback, &recursionGuard, &layers, maxRecursion, this ]( const QgsVectorLayer * vLayer, bool onlySaveSelected, auto &&findReferenced ) -> void
2022-08-11T15:16:39.9189339Z                                                                                  ~~^~~~~~~~~~~~
2022-08-11T15:16:39.9190151Z /root/QGIS/src/analysis/processing/qgsalgorithmpackage.cpp:207:85: error: lambda capture 'maxRecursion' is not required to be captured for this use [-Werror,-Wunused-lambda-capture]
2022-08-11T15:16:39.9191173Z       const auto findReferencing = [ &project, &feedback, &recursionGuard, &layers, maxRecursion, this ]( const QgsVectorLayer * vLayer, bool onlySaveSelected, auto &&findReferencing ) -> void
2022-08-11T15:16:39.9191786Z                                                                                   ~~^~~~~~~~~~~~
2022-08-11T15:16:39.9192069Z 2 errors generated.
```

local gcc build is fine.

--- a/src/analysis/processing/qgsalgorithmpackage.cpp
+++ b/src/analysis/processing/qgsalgorithmpackage.cpp
@@ -135,7 +135,7 @@ QVariantMap QgsPackageAlgorithm::processAlgorithm( const QVariantMap &parameters
       int recursionGuard { 0 };
 
       // This function recursively finds referenced layers
-      const auto findReferenced = [ &project, &feedback, &recursionGuard, &layers, maxRecursion, this ]( const QgsVectorLayer * vLayer, bool onlySaveSelected, auto &&findReferenced ) -> void
+      const auto findReferenced = [ =, &project, &feedback, &recursionGuard, &layers ]( const QgsVectorLayer * vLayer, bool onlySaveSelected, auto &&findReferenced ) -> void
       {
         const QgsVectorLayer *originalLayer { qobject_cast<QgsVectorLayer *>( project->mapLayer( mClonedLayerIds.value( vLayer->id(), vLayer->id() ) ) ) };
         Q_ASSERT( originalLayer );
@@ -204,7 +204,7 @@ QVariantMap QgsPackageAlgorithm::processAlgorithm( const QVariantMap &parameters
       };
 
       // This function recursively finds referencing layers
-      const auto findReferencing = [ &project, &feedback, &recursionGuard, &layers, maxRecursion, this ]( const QgsVectorLayer * vLayer, bool onlySaveSelected, auto &&findReferencing ) -> void
+      const auto findReferencing = [ =, &project, &feedback, &recursionGuard, &layers ]( const QgsVectorLayer * vLayer, bool onlySaveSelected, auto &&findReferencing ) -> void
       {
         const QgsVectorLayer *originalLayer { qobject_cast<QgsVectorLayer *>( project->mapLayer( mClonedLayerIds.value( vLayer->id(), vLayer->id() ) ) ) };
         Q_ASSERT( originalLayer );

